packages/ui/libs/ui/theme/src/providers/themeModeProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, createContext, useContext, useState } from 'react';
+import {
+  getThemeModeFromLS,
+  saveThemeModeToLS,
+} from '@energyweb/origin-ui-shared-state';
+
+export type ThemeModeType = 'dark' | 'light';
+
+const ThemeModeStore = createContext<ThemeModeType>(null);
+const ThemeModeDispatch = createContext<() => void>(null);
+
+export const ThemeModeProvider: FC = ({ children }) => {
+  const [themeMode, setThemeMode] = useState<ThemeModeType>(
+    getThemeModeFromLS() || 'dark'
+  );
+  const setMode = () => {
+    const newMode = themeMode === 'dark' ? 'light' : 'dark';
```

Comment:
Using an enum instead of strings `'dark'` and `'light'`would help avoid typos and make it more type-safe.